x-pack/legacy/plugins/epm/server/datasources/create.ts

```diff
@@ -17,30 +23,29 @@ import * as Registry from '../registry';
 
 export async function createDatasource(options: {
   savedObjectsClient: SavedObjectsClientContract;
-  pkgkey: string;
+  payload: DatasourcePayload;
```

Comment:
Can you replace `payload` with its three members? So, `createDatasource` will accept 5 items instead of 3.